### PR TITLE
[@reach/router v1.2.1] Fixed return type of navigate function

### PR DESCRIPTION
**Pull request by awwit**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/reach/router/blob/master/src/lib/history.js#L48-L70>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by awwit** [comment]
@kingdaro @xMokAx Guys, please see the corrections.

File: types/reach__router/reach__router-tests.tsx [hunk 1]
```diff
@@ -40,7 +41,7 @@ render(
             {context => (
                 <>
                     <div>hostname is {context.location.hostname}</div>
-                    <button onClick={() => context.navigate('/')}>
+                    <button onClick={(): Promise<void> => context.navigate('/')}>
```
Comment (kingdaro) [inline types/reach__router/reach__router-tests.tsx]: This return annotation shouldn't be necessary

File: types/reach__router/reach__router-tests.tsx [hunk 2]
```diff
@@ -50,7 +51,7 @@ render(
...
```
Comment (kingdaro) [inline types/reach__router/reach__router-tests.tsx]: same here

File: types/reach__router/index.d.ts [hunk 3]
```diff
@@ -89,7 +91,10 @@ export interface MatchRenderProps<TParams> {
 
 export class Match<TParams> extends React.Component<MatchProps<TParams>> {}
 
-export type NavigateFn = (to: string, options?: NavigateOptions<{}>) => void;
+export interface NavigateFn {
+    (to: string, options?: NavigateOptions<{}>): Promise<void>;
+    (to: number): Promise<void>;
```
Comment (kingdaro) [inline types/reach__router/index.d.ts]: could you point to the section in the docs showing where navigate can accept a number? does it also accept options in this case?

**Review by itsMapleLeaf — CHANGES_REQUESTED** [review]

File: types/reach__router/index.d.ts (hunk 3 above)
Comment (awwit) [reply types/reach__router/index.d.ts]: Here: https://github.com/reach/router/blob/master/src/lib/history.js#L49-L51

File: types/reach__router/reach__router-tests.tsx (hunk 1 above)
Comment (awwit) [reply types/reach__router/reach__router-tests.tsx]: A must, as this is a test to check the return value.

File: types/reach__router/index.d.ts (hunk 3 above)
Comment (kingdaro) [reply types/reach__router/index.d.ts]: 🤔 not sure if I like supporting undocumented features, but I guess this is fine

**Review by itsMapleLeaf — APPROVED** [review]

File: types/reach__router/index.d.ts (hunk 3 above)
Comment (awwit) [reply types/reach__router/index.d.ts]: By the way, there is a mention of this in the documentation :)
https://github.com/reach/router/blob/master/website/src/markdown/api/navigate.md#to

**Pull request merged**